abci/types/mocks/base.go

```diff
@@ -0,0 +1,182 @@
+package mocks
```

Comment:
Could we get a few doc comments in this package? From reading the code I can see what it's for—and the behaviour makes sense. But it's not obvious why this is a mock rather than a straightforward delegating implementation (I assume this is mainly because it's using the other test plumbing in the same package).